### DISCUSSION TITLE
Scripting >E var Dimmer not found

### DISCUSSION THREAD

**Discussion by martin072** [arendst/Tasmota#14397] [Q&A] [answered]
I am using the dimmer script for a Sonoff D1 and works fine except for the following part:

```
>E
slid=Dimmer
if chg[slid]>0 {
   dval=slid 
}
```

gives the following on the console
```
13:01:20.399 var not found: Dimmer
```

Any ideas?

Edit:
I guess the reason is that all RESULTS come into >E, so a POWER on/off will not include the Dimmer variable, that is probably causing the message on the console?

**Comment by thxthx0** [comment]
> Any ideas?

Hi, 
https://gist.github.com/thxthx0/c22555f6ec25c3367264729a0ff74dce#gistcomment-3977645

**Reply by martin072** [reply]
HI, Thanks for your reply, I had already compiled with the #define SCRIPT_POWER_SECTION , so that's not it. 
As I mentioned when a POWER event occurs, it does not contain the Dimmer value.. 

```
11:56:55.446 MQT: home/sonoffs/stat/ledbar/RESULT = {"POWER":"ON"}
```

**Reply by thxthx0** [reply]
Hi, that's correct, by default it does not contain the Dimmer value
when a POWER event occurs, only when the slider is moved,

but in my view the above error (var not found: Dimmer) 
only occurs if SCRIPT_POWER_SECTION is missing
or if no PWM is defined in the module configuration.

**Comment by martin072** [comment, marked as answer]
Seems you're correct. re-flashed with SCRIPT_POWER_SECTION enabled and that seems to work.
